### PR TITLE
Making master services always restart, plus fixes

### PR DESCRIPTION
**Pull request by Cynerva**
This builds on the work done by @hyperbolic2346 in https://github.com/juju-solutions/kubernetes/pull/165, with a couple fixes on top that I think are needed. Cheers.

### CODE REVIEW

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py [hunk 1]
```diff
@@ -945,6 +960,7 @@ def update_nrpe_config(unused=None):
     current_unit = nrpe.get_nagios_unit_name()
     nrpe_setup = nrpe.NRPE(hostname=hostname)
     nrpe.add_init_service_checks(nrpe_setup, services, current_unit)
+    nrpe.add_init_service_checks(nrpe_setup, master_services, current_unit)
```
Comment (kwmonroe) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: why are we nrpe'ing both snap.*.daemon and *?

Comment (hyperbolic2346) [reply cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: I came to comment on this. Originally I was merging nrpe into the master_services array and ended up pulling it for time. I guess I didn’t pull it all, but this should be nuked.

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py [hunk 2]
```diff
@@ -1060,10 +1076,8 @@ def shutdown():
     """ Stop the kubernetes master services
 
     """
-    service_stop('snap.kube-apiserver.daemon')
-    service_stop('snap.kube-controller-manager.daemon')
-    service_stop('snap.kube-scheduler.daemon')
-    service_stop('snap.kube-proxy.daemon')
+    for service in master_services:
+        service_stop('snap.%s.daemon' % service)
```
Comment (hyperbolic2346) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: It originally said snap.*.daemon and I am happy with this, but the service_stop comments claim %s only. We should clarify the called function or maybe I was just looking at the wrong thing as it was initd stuff.

File: cluster/juju/layers/kubernetes-master/templates/service-always-restart.conf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[Unit]
+StartLimitIntervalSec=0
+
+[Service]
+RestartSec=1
```
Comment (hyperbolic2346) [inline cluster/juju/layers/kubernetes-master/templates/service-always-restart.conf]: 1 second seems a little spammy, no?

Comment (Cynerva) [reply cluster/juju/layers/kubernetes-master/templates/service-always-restart.conf]: Not to me, I like the idea that it'll restart quickly. But this is super subjective so it's hard to have a meaningful debate about it. If you still like 15sec then I'll revert it to that.

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py (hunk 1 above)
Comment (Cynerva) [reply cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: Ah, I missed this. Pulled it. Thanks!

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py (hunk 2 above)
Comment (Cynerva) [reply cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: This is calling `charmhelpers.core.host.service_stop`, which calls `systemctl stop <service>`, which complains if you omit the `snap.` and `.daemon` bits:
```
$ sudo systemctl stop kube-proxy
Failed to stop kube-proxy.service: Unit kube-proxy.service not loaded.
```

I'm pretty sure we need it.

File: cluster/juju/layers/kubernetes-master/templates/service-always-restart.conf (hunk 3 above)
Comment (Cynerva) [reply cluster/juju/layers/kubernetes-master/templates/service-always-restart.conf]: I went for 10 sec. Cheers.

**Comment by Cynerva** [comment]
Revised:
* Removed the NRPE thing
* Bumped RestartSec up to 10

Anything else?

**Pull request merged**